### PR TITLE
Mini editorial for IOI 2013 Game

### PR DESCRIPTION
**Pull request by dolphingarlic**
Closes https://github.com/cpinitiative/usaco-guide/issues/916. There's not a lot to say about this problem...

### CODE REVIEW

**Comment by bqi343** [comment]
segment tree of sparse segment trees was intended though (using the optional note here - https://usaco.guide/plat/sparse-seg?lang=cpp#solution)

File: solutions/IOI 13-game.mdx [hunk 1]
```diff
@@ -28,6 +28,15 @@ because they support point updates and range queries. Each segment tree node
 stores a treap, and updating a node involves changing $\mathcal O(\log C)$ values
 in its treap (similar to updating a 2D segment tree node).
 
+<Optional>
+
+It's possible (and probably intended) to solve this problem using a segment trees
```
Comment (bqi343) [inline solutions/IOI 13-game.mdx:33]: It \*is\* intended. I'm pretty sure I remember reading about this, though I can't find it ...

Preferably include such a solution

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/IOI 13-game.mdx [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+ll calculate(int P, int Q, int U, int V) {
+	return segtree.query(P, U, Q, V);
+}
+```
+
+## Approach 2 - Memory-optimized 2D sparse segment tree
+
+Although the previous approach is somewhat simpler, this approach was intended,
+and involves optimizing the memory usage of a sparse segment tree from 
+$\mathcal O(N \log N)$ to $\mathcal O(N)$.
```
Comment (bqi343) [inline solutions/IOI 13-game.mdx:217]: Wouldn't it be N log(max value), not N log N?

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by dolphingarlic** [comment]
@bqi343 bump

File: solutions/IOI 13-game.mdx [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+each node stores the GCD of its two children. Since $R$ can be quite big, this
+needs to be a sparse segment tree; another alternative would be a balanced binary
+tree like a treap.
+
+However, a sparse 2D segment tree uses *just* a bit too much memory, and only 
+scores 80 points. Fortunately for us, there are two ways to get around this!
+
+## Approach 1 - Sparse segment tree of BBSTs
+
+Although BBSTs use 4 times less memory than segment trees, a BBST of BBSTs is
```
Comment (bqi343) [inline solutions/IOI 13-game.mdx:24]: > a BBST of BBSTs is not possible

er actually range trees are BBSTs of BBSTs https://en.wikipedia.org/wiki/Range_tree

if the tree gets too unbalanced after a certain # of insertions then you end up rebuilding an entire subtree I think

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Review by bqi343 — APPROVED** [review]

**Pull request merged**